server/src/main/java/org/opensearch/rest/action/cat/RestPitSegmentsAction.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.cat;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateRequest;
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.admin.indices.segments.IndexSegments;
+import org.opensearch.action.admin.indices.segments.IndexShardSegments;
+import org.opensearch.action.admin.indices.segments.IndicesSegmentResponse;
+import org.opensearch.action.admin.indices.segments.PitSegmentsAction;
+import org.opensearch.action.admin.indices.segments.PitSegmentsRequest;
+import org.opensearch.action.admin.indices.segments.ShardSegments;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.common.Strings;
+import org.opensearch.common.Table;
+import org.opensearch.index.engine.Segment;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestActionListener;
+import org.opensearch.rest.action.RestResponseListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+/**
+ * Rest action for pit segments
+ */
+public class RestPitSegmentsAction extends AbstractCatAction {
+    @Override
+    public List<RestHandler.Route> routes() {
+        return unmodifiableList(asList(new Route(GET, "/_cat/pit_segments"), new Route(GET, "/_cat/pit_segments/{pit_id}")));
+    }
+
+    @Override
+    public String getName() {
+        return "cat_pit_segments_action";
+    }
+
+    @Override
+    public boolean allowSystemIndexAccessByDefault() {
+        return true;
+    }
+
+    @Override
+    protected BaseRestHandler.RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
+        final String[] pitIds = Strings.splitStringByCommaToArray(request.param("pit_id"));
+        final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
+        clusterStateRequest.local(false);
+        clusterStateRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterStateRequest.masterNodeTimeout()));
+        clusterStateRequest.clear().nodes(true).routingTable(true).indices("*");
+
+        return channel -> client.admin().cluster().state(clusterStateRequest, new RestActionListener<>(channel) {
```

Comment:
Because we need cluster state call within the rest layer, can't avoid this. Which means, PIT segments API will also need cluster state permissions to work.

Reply:
Why `PitSegmentsAction` needs cluster state permission? The `/_cat/...` certainly does, but not `PitSegmentsAction` I believe

Reply:
Since the cluster state is used in the table logic, we are making the cluster state api call here which makes it a dependency permission

Reply:
That's the `/_cat/...` API

Reply:
Yeah only for` /_cat/...` API i meant that there is dependency. SegmentsAction by itself is not dependant on this.